File: year2/MPP/C# Client Server Sockets/Proiect_MPP_C#_VARIANTA_FINALA/DomaninRepo/validators/ValidatorDonatii.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain_Repo.entities;

namespace DomainRepoLibrary.validators
{
    public class ValidatorDonatii
    {
        private string Errors;
        public void valideaza(Donatie d)
        {
            Errors = "";
        if (d.sumaDonata <= 0F)
            Errors += "suma invalida!";
            if (Errors.Length > 0)
                throw new Exception(Errors);
    }
}
}

File: year2/MPP/Java Client Server Sockets/SablonJavaDB/src/main/java/me/validators/ValidatorDonatii.java
package me.validators;

import me.entities.Donatie;
import org.controlsfx.validation.ValidationMessage;

import javax.xml.bind.ValidationException;

public class ValidatorDonatii {
    private String errors;
    public void valideaza(Donatie d) throws ValidationException {
        errors = "";
        if (d.getSumaDonata() <=0F)
            errors += "suma invalida!";
        if(errors.length() > 0)
            throw new ValidationException(errors);
    }
}
